### PR TITLE
Enable and fix a number of lints

### PR DESCRIPTION
**Pull request by kevmoo**
Also make a number of fields final

### CODE REVIEW

**Comment by kevmoo** [comment]
Friendly ping?

**Comment by pulyaevskiy** [comment]
Thanks Kevin! Will take a look tonight

File: lib/corsac_jwt.dart [hunk 1]
```diff
@@ -109,33 +110,39 @@ class JWTRsaSha256Signer implements JWTSigner {
...
-int _secondsSinceEpoch(DateTime dateTime) {
-  return (dateTime.millisecondsSinceEpoch / 1000).floor();
+Map<String, T> _decode<T>(String input) {
+  try {
+    return Map.from(
+      _jsonToBase64Url.decode(_base64Padded(input)) as Map<String, dynamic>,
+    );
+  } catch (e) {
+    throw JWTError('Could not decode token string. Error: $e.');
+  }
```
Comment (pulyaevskiy) [inline lib/corsac_jwt.dart:139]: Thanks for adding this!

File: lib/corsac_jwt.dart [hunk 2]
```diff
@@ -201,22 +207,14 @@ class JWT {
...
-      _headers = Map<String, String>.from(
-          _jsonToBase64Url.decode(_base64Padded(encodedHeader)));
-      _claims = Map<String, dynamic>.from(
-          _jsonToBase64Url.decode(_base64Padded(encodedPayload)));
-    } catch (e) {
-      throw JWTError('Could not decode token string. Error: ${e}.');
-    }
-  }
+  JWT._(this.encodedHeader, this.encodedPayload, this.signature)
+      : headers = Map.unmodifiable(_decode(encodedHeader)),
```
Comment (pulyaevskiy) [inline lib/corsac_jwt.dart:211]: Thanks for `unmodifiable`.

**Review by pulyaevskiy — APPROVED** [review]
LGTM, thanks!

**Pull request merged**